lib/widgets/scaffold.dart

```diff
@@ -163,4 +167,58 @@ class AppScaffold extends StatelessWidget {
       ),
     );
   }
+
+  ///
+  /// Build content body based on is web version.
+  /// When web version ad an action bar to right side.
+  ///
+  Widget buildBody(BuildContext context, {bool enableContent = false}) {
+    final List<Widget> actionButtons =
+        !kIsWeb // check to avoid building action button twice in mobile
+            ? List.empty()
+            : ([leading] + actions)
+                .map((action) => buildMaterialActionButton(action, context,
+                    enableContent: enableContent))
+                .where((element) => element != null)
+                .toList();
+
+    return kIsWeb && actionButtons.isNotEmpty
+        ? Stack(
+            children: [
+              Center(
+                  child: ConstrainedBox(
+                      constraints: BoxConstraints(maxWidth: maxBodyWidth),
+                      child: body)),
+              ActionSideBar(actionButtons),
+            ],
+          )
+        : body;
```

Comment:
The variable and multiple conditions make this a bit hard to read. How about this:
```suggestion
    return !kIsWeb
        ? body
        : Stack(
            children: [
              Center(
                  child: ConstrainedBox(
                      constraints: BoxConstraints(maxWidth: maxBodyWidth),
                      child: body)),
              ActionSideBar(
                  ([leading] + actions)
                    .map((action) => buildMaterialActionButton(action, context,
                        enableContent: enableContent))
                    .where((element) => element != null)
                    .toList(),
              ),
            ],
          );
```

Reply:
Still want to check if there is action to be added to sidebar, otherwise sidebar shouldn't be added.

Reply:
Moved mapping through actions:
https://github.com/student-hub/acs-upb-mobile/pull/250/files#diff-2804fa2a5ae6726cac5690433203db74702f6ee38877cb902a9d5095c419b578R175-R176
https://github.com/student-hub/acs-upb-mobile/pull/250/files#diff-2804fa2a5ae6726cac5690433203db74702f6ee38877cb902a9d5095c419b578R175-R176

Reply:
I would argue that even if there are no actions to put in the sidebar, the maxWidth constraint should still be applied on web.

So I still think you should use my proposed code structure, and inside the sidebar check if the list of actions is empty; if it is, simply don't build anything.

Reply:
Right, missed the part that if action list is empty max width constraint won't be applied , thanks.

Reply:
https://github.com/student-hub/acs-upb-mobile/pull/250/commits/dad30a96ea179228206a08abb53d168a6ef85933